### PR TITLE
Change default fallback encoding for GetEncoding in Start-Transcript

### LINKED ISSUE #13677: Start-Transcript uses lossy ASCII character encoding instead of BOM-less UTF-8 if the target file happens to exist
Note: 
 * Windows PowerShell is _not_ affected - UTF-8 with BOM is used consistently there.
 * A fix for the closely related #13678 could probably encompass a fix for this issue as well.

When the target file _happens to exist and doesn't have a BOM_ - even without `-Append` - `Start-Transcript` mistakenly uses ASCII encoding to write the file.

Without `-Append`, `Start-Transcript` shouldn't even look at the existing file - it should simply replace it, and use the default encoding (BOM-less UTF-8). See #13678 for a closely related bug _with_ `-Append`.

Instead, `Start-Transcript` apparently looks for a BOM at the start of an existing file and uses ASCII encoding if it doesn't find one.

## Steps to reproduce

```powershell
# To surface the bug, make sure that the target file exists and doesn't have a BOM.
$null > temp:/$PID.txt

$null = Start-Transcript temp:/$PID.txt
'ü' # output a string with a non-ASCII-range character
$null = Stop-Transcript

Select-String -Quiet 'ü' temp:/$PID.txt | Should -BeTrue

Remove-Item temp:/$PID.txt
```

## Expected behavior

The test should succeed.

## Actual behavior

The test fails, because `ü` was transliterated to verbatim `?`, suggesting that ASCII encoding was used to write the file.

## Environment data

```none
PowerShell Core 7.1.0-preview.7
```

### LINKED ISSUE #13678: Start-Transcript -Append uses lossy ASCII character encoding when appending to a BOM-less target file
Note: Unlike #13677, Windows PowerShell exhibits the same problem.

`Start-Transcript -Append` tries to _match_ the target file's existing character encoding.

As an aside: The built-in cmdlets are inconsistent with respect to this behavior: `Out-File -Append` does _not_ match the existing encoding, whereas `Export-Csv -Append`, `Start-Transcript -Append`, and `Add-Content` do.

In the _absence_ of a BOM in the target file, ASCII encoding is currently assumed, which results in the loss of non-ASCII-range characters, as they are transliterated to verbatim `?`.

UTF-8 (by definition BOM-less, since a file is being _appended_ to) should be assumed and used instead.

## Steps to reproduce

```powershell
# To surface the bug, make sure that the target file exists and doesn't have a BOM.
'ä' > temp:/$PID.txt

# Start the transcript and *append* to the existing file:
$null = Start-Transcript -Append temp:/$PID.txt
'ü' # output a string with a non-ASCII-range character
$null = Stop-Transcript

Select-String -Quiet 'ü' temp:/$PID.txt | Should -BeTrue

Remove-Item temp:/$PID.txt
```

## Expected behavior

The test should succeed.

## Actual behavior

The test fails, because `ü` was transliterated to verbatim `?`.

## Environment data

```none
PowerShell Core 7.1.0-preview.7
```

### PR DESCRIPTION
**Pull request by Gimly**
# PR Summary

The System.Management.Automation.Utils.GetEncoding method should return UTF8 without BOM as a default but was returning ASCII.

This is fixed by returning an instance of the Utf8Encoding with the BOM set to false as a default fallback encoding.

## PR Context

This should fix #13678 and fix partially #13677

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by rjmholt** [comment]
@PowerShell/powershell-committee please review this potentially breaking change

File: src/System.Management.Automation/engine/Utils.cs [hunk 1]
```diff
@@ -1458,7 +1458,7 @@ internal static Encoding GetEncoding(string path)
                 return Encoding.Unicode;
             }
 
-            return Encoding.ASCII;
+            return new UTF8Encoding(false);
```
Comment (rjmholt) [inline src/System.Management.Automation/engine/Utils.cs:1461]: Please label the boolean argument here

Comment (iSazonov) [reply src/System.Management.Automation/engine/Utils.cs:1461]:
```suggestion
            return new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
```

**Comment by iSazonov** [comment]
For PowerShell-Committee:
- it affects only transcribe output files.
- we already changed defaults to Uft8NoBOM in other code paths
- I think it is a bug and we shouldn't consider this as a breaking change - it is unexpected behavior if we change encoding of existent file.

**Comment by SteveL-MSFT** [comment]
ASCII being a subset of UTF8 makes this an unlikely impactful breaking change (bucket 3) for me.  Seems like the right thing to do.

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee reviewed this, we agree that this is a bucket 3 breaking change that is unlikely to have negative impact on anyone.  This should have been changed along with the other encoding changes to UTF-8 by default and is a bug fix.

File: src/System.Management.Automation/engine/Utils.cs [hunk 2]
```diff
@@ -1458,7 +1458,7 @@ internal static Encoding GetEncoding(string path)
                 return Encoding.Unicode;
             }
 
-            return Encoding.ASCII;
+            return new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/Utils.cs:1461]: I guess you can use the static `utf8NoBom` declared below:
```
internal static readonly UTF8Encoding utf8NoBom =
            new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
```

Comment (rjmholt) [reply src/System.Management.Automation/engine/Utils.cs:1461]: @Gimly please apply this change

Comment (Gimly) [reply src/System.Management.Automation/engine/Utils.cs]: Done, you're right I hadn't seen the constant.

**Review by rjmholt — APPROVED** [review]

**Comment by rjmholt** [comment]
@daxian-dbw please re-review if you can

**Review by iSazonov — APPROVED** [review]

**Comment by Gimly** [comment]
@rjmholt I'm not sure about those CI fails, do I need to do something about them?

**Comment by iSazonov** [comment]
I restarted CI Windows.

**Comment by iSazonov** [comment]
@Gimly Thanks for your contribution!

**Review by Steg17 — COMMENTED** [review]
:)

**Pull request merged**